pkg/controllers/deploymentreplicassyncer/deployment_replicas_syncer_controller.go

```diff
@@ -98,69 +106,112 @@ func (r *DeploymentReplicasSyncer) Reconcile(ctx context.Context, req controller
 	binding := &workv1alpha2.ResourceBinding{}
 	bindingName := names.GenerateBindingName(util.DeploymentKind, req.Name)
 
-	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: req.Namespace, Name: bindingName}, binding); err != nil {
-		if apierrors.IsNotFound(err) {
-			klog.Infof("no need to update deployment replicas for binding not found")
+	err := wait.PollUntilContextTimeout(ctx, waitDeploymentStatusInterval, waitDeploymentStatusTimeout, true,
```

Comment:
I think using `PollUntilContextTimeout` in the `controller` is not a very good solution. The controller should `watch` for these `changes `to do the corresponding `Reconcile`, instead of constantly `querying` for these changes.

Reply:
+1,

The controller might need to wait for an event that it doesn't watch, in that case, it can re-queue the resource key.